frame/support/src/traits.rs

```diff
@@ -653,9 +653,15 @@ bitmask! {
 	}
 }
 
+/// Trait to deals with time.
 pub trait Time {
+	/// Type for the time.
 	type Moment: SimpleArithmetic + Parameter + Default + Copy;
 
+	/// Duration for a Julian year (365.25 days).
+	fn year() -> Self::Moment;
+
+	/// Return current moment.
 	fn now() -> Self::Moment;
```

Comment:
Now you are mixing some time related functionality (`year()`) with a `moment` that can be anything. 

I still think we should keep this `Time` trait simple and say that `now()` will return a `moment` that increases between blocks, but without specifying what it is. It could also be the moment from the genesis of the chain.

I would add another trait `UnixTime` that clearly says that `now()` returns the `Duration` from the unix epoch. I would directly return `core::time::Duration`. This already brings all the functions for converting between the different time formats.

The timestamp module would implement both of these traits and return the same value for both `now()` calls, but as different types. `Staking` would require that `Time` implements `UnixTime`.

Reply:
yes, I completely agree